Feature/init cluster at startup

jsvisa opened the pull request.

jsvisa committed:
fix(make): git describe doesn't provide tag

jsvisa committed:
refactor(heketi-start): ascii sort command list

jsvisa committed:
feat(docker): add heketi-init to load cluster at startup

jsvisa commented:
@ijumps  PTAL

ijumps commented on Dockerfile:
```diff
@@ -12,6 +12,7 @@ RUN apt-get update \
 ADD ./client/cli/go/heketi-cli /usr/bin/heketi-cli
 ADD ./heketi-start.sh /usr/bin/heketi-start.sh
+ADD ./heketi-init.sh /usr/bin/heketi-init.sh
 VOLUME [/etc/heketi, /var/lib/heketi]
```
`[/etc/heketi, /var/lib/heketi]` -> `["/etc/heketi", "/var/lib/heketi"]`

jsvisa replied:
@ijumps 这个双引号现在是强制的吗？

ijumps commented:
`exec "/usr/bin/heketi --config=$config"` should be `exec "/usr/bin/heketi" "--config=$config"` in file `heketi-start.sh`

ijumps commented on heketi-init.sh:
```diff
@@ -0,0 +1,70 @@
+        echo "loading $endpoint"
+    fi
+
+    printf "{\"endpoints\": [\n\t{\"nodes\": [" > $volume_path/tmp-topology.json
```
`endpoints`  -> `clusters`

ijumps replied:
ref: https://github.com/heketi/heketi/blob/master/client/cli/go/topology-sample.json

jsvisa replied:
@ijumps fixed

jsvisa commented:
@ijumps IMO, the previous is correct, the command and argument should be an entirety, then passed to `exec`

> exec "/usr/bin/heketi --config=$config" should be exec "/usr/bin/heketi" "--config=$config" in file heketi-start.sh

jsvisa committed:
fix(*): see in detail

1. volume name should be surround with '"'
2. the correct format in exec is 'exec "command" "argument"'

ijumps commented:
LGTM
